### PR TITLE
Fix Spyfakku

### LINKED ISSUE #4263: Spy.fakku 1.4.3 not working
### Source information

Spy.fakku 1.4.3

### Source language

English 

### Steps to reproduce

Open extension and or do normal search. Extension does not display any titles. Click on extension alone, open webview and works fine. 

### Expected behavior

To display titles via search, and open titles to read or save ro lib.

### Actual behavior

The extension does not display anything but a error. However webview still works fine.
Screenshot_20240726_083055_TachiyomiJ2K

### Mihon/Tachiyomi version

Tachij2k 1.7.4

### Android version

Android 14

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by KenjieDec**
Closes #4263
Hello, sorry if the code's messy, please tell me if there's anything that can be shortened, thank you!

Checklist:
- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt [hunk 1]
```diff
@@ -45,28 +53,77 @@ class SpyFakku : HttpSource() {
         .set("Origin", baseUrl)
 
     override fun popularMangaRequest(page: Int): Request {
-        return GET("$baseApiUrl/library?sort=released_at&page=$page", headers)
+        return GET("$baseUrl$baseApiEnd?sort=released_at&page=$page", headers)
     }
 
     override fun popularMangaParse(response: Response): MangasPage {
         val library = response.parseAs<HentaiLib>()
+        val data = getMangasDatas(library.data)
```
Comment (AwkwardPeak7) [inline src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:61]: image
response is like this which doesn't seem to match your parsing, maybe they changed it? or am I missing something?

File: src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt [hunk 2]
```diff
@@ -45,28 +53,77 @@ class SpyFakku : HttpSource() {
...
-        val mangas = library.archives.map(::popularManga)
+        val mangas = data.first.map(::popularManga)
 
-        val hasNextPage = library.archives.isNotEmpty()
+        val hasNextPage = data.second
 
         return MangasPage(mangas, hasNextPage)
     }
 
+    private fun getMangasDatas(data: List<JsonElement>): Pair<List<ShortHentai>, Boolean> {
```
Comment (AwkwardPeak7) [inline src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:70]: you could make a custom serializer but that can be a bit complicated

File: src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt (hunk 1 above)
Comment (KenjieDec) [reply src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:61]: Hello, I added `.substringAfter("\n")` in `Response.parseAs()`  function to get only line 2's object. Their source seem to do the same.

File: src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt (hunk 2 above)
Comment (KenjieDec) [reply src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:70]: Hello, Thanks for the suggestion! 
Sorry, but could you please explain more on how to do that?

Comment (AwkwardPeak7) [reply src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:70]: https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/serializers.md#custom-serializers

tho it is fine without it too

Comment (KenjieDec) [reply src/en/spyfakku/src/eu/kanade/tachiyomi/extension/en/spyfakku/SpyFakku.kt:70]: Hello, thanks for linking the explanation! 
Sorry, but I think it's too complex for me. 🙏

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**